documentation/modules/ROOT/pages/02-architecture.adoc

```diff
@@ -1,104 +1,255 @@
+** https://cloudevents.io/[CloudEvents] &mdash; Provides a standard for event metadata
+** Rust and https://access.redhat.com/products/quarkus[Quarkus] &mdash; Implementation examples.
+** https://opentelemetry.io/[OpenTelemetry] &mdash; (Optional) Facilitates tracing for observability.
 
 [#in_depth]
 == An in-depth look at the solution's architecture
 
-Technical description including all or some of the following: architecture ir ed diagrams. In-depth details of the decisions made and solutions used. Description of each service and what it is used for. Description of any integration.
+Building applications around the Event Mesh is a solution that can be applied both to existing and new projects.
+For existing applications, domain boundaries can guide the division into modular components, which may evolve into separate services.
+These modules will generate commands intended for the _Event Mesh_, which will then route these events to their designated endpoints.
+The _Event Mesh_ will allow for such transition, gradually making the system not only more responsive but also better suited to the real-world business logic.
+
+=== Problem Analysis
+
+Traditional systems often enforce strict transactional consistency, which can impede application performance and compromise business logic.
+For instance, upon the completion of a ride-sharing service at the end-user's destination, the system should reliably capture this real-world event.
+The capture should be performed regardless of any potential operational disruptions affecting dependent services (e.g., invoicing).
+
+In such scenarios, transactional applications typically return an error, which prevents any data from being changed, and causes the loss of real-world intent of end-users.
+This results in an adverse user experience and a deviation from the genuine business process, potentially leading to customer dissatisfaction.
+
+=== Solution Breakdown
+
+The concept of employing the _Event Mesh_ as a central, reliable hub for dispatching commands, as events, lies at the heart of this solution.
+This approach aligns closely with the Command Query Responsibility Segregation (_CQRS_) pattern, which distinctly categorizes commands and queries.
+Commands, in this context, are modeled as asynchronous events, designed to undergo processing by the _Event Mesh_.
+On the other hand, queries are synchronous operations, safe to retry, ensuring no loss of data integrity due to transient errors.
+
+The primary responsibility of the _Event Mesh_ is twofold.
+Firstly, it persists the incoming events, thereby maintaining a record of changes in the system's state.
+Secondly, it routes these events to their respective endpoints, ensuring that the appropriate microservices are notified and can subsequently update their internal states based on the event data.
+
+The mesh's inherent resilience is further bolstered by its exponential backoff strategy, which it employs when encountering operational failures.
```

Comment:
why is an _exponential_ backoff strategy benefitial and not only one that does retries after a fixed amount of time ? This could be added here, as otherwise I would leave out _exponential backoff_ and replace it with just retries to keep it more simple.

Reply:
Maybe I could add a sentence or two about exponential backoff - it gives the overwhelmed endpoint time to recover, which in many cases allow system to recover without intervention.

Reply:
I think just retries is good enough at this level.

linear / exponential are more implementation details - which have their pros/cons. We could (later?) link to the knative page: https://knative.dev/docs/eventing/event-delivery/#configuring-subscription-event-delivery  but it also does not really well explain _why_ for each